release/src/milestones.ts

```diff
@@ -196,3 +265,313 @@ export async function setMilestoneForCommits({
   }
 }
 
+const issueLink = (
+  { owner, repo, issueNumber }:
+  { owner: string, repo: string, issueNumber: number }
+) => `https://github.com/${owner}/${repo}/issues/${issueNumber}`;
+
+export async function checkMilestoneForRelease({
+  github,
+  owner,
+  repo,
+  version,
+  commitHash,
+}: GithubProps & { version: string, commitHash: string }) {
+  const releaseMilestone = await findMilestone({ github, owner, repo, version });
+
+  if (!releaseMilestone) {
+    throw new Error(`No open milestone found for ${version}`);
+  }
+
+  const closedMilestoneIssues = await getMilestoneIssues({
+    github, owner, repo, version, state: 'closed', milestoneStatus: 'open',
+  });
+  const openMilestoneIssues = await getMilestoneIssues({
+    github, owner, repo, version, state: 'open', milestoneStatus: 'open',
+  });
+
+  const lastTag = await getLastReleaseTag({
+    github,
+    owner,
+    repo,
+    version,
+  });
+
+  const compareResponse = await github.rest.repos.compareCommitsWithBasehead({
+    owner,
+    repo,
+    basehead: `${lastTag}...${commitHash}`,
+  });
+
+  const commits = compareResponse.data.commits;
+
+  console.log(`Found ${commits.length} commits in release branch`);
+  console.log(`Found ${closedMilestoneIssues.length} issues in milestone`);
+
+  const milestoneIssueSet = new Set(closedMilestoneIssues.map(issue => issue.number));
+  const commitIssueSet = new Set<number>();
+
+  // make sure every commit in the release branch has a corresponding issue in the milestone
+  const commitIssueMap: Record<string, number[]> = {};
+  const issueCommitMap: Record<number, string> = {};
+
+  for (const commit of commits) {
+    const prNumbers = getPRsFromCommitMessage(commit.commit.message);
+    if (!prNumbers) {
+      console.log('No PRs found in commit message', commit.commit.message);
+      continue;
+    }
+
+    const issueNumbers: number[] = [];
+
+    for (const prNumber of prNumbers) {
+      if (issueNumbers.includes(prNumber)) {
+        continue;
+      }
+      issueNumbers.push(...(await getOriginalIssues({
+        github,
+        owner,
+        repo,
+        issueNumber: prNumber,
+      })));
+    }
+
+    const uniqueIssues = _.uniq(issueNumbers.filter(isNotNull));
```

Comment:
I don't mind, but sometimes it's new Set, sometimes underscore

Reply:
sometimes I want a set, sometimes I want an array 🤷